slackclient/client.py

```diff
@@ -88,27 +91,29 @@ def api_call(self, method, timeout=None, **kwargs):
         response_body = self.server.api_call(method, timeout=timeout, **kwargs)
         try:
             result = json.loads(response_body)
-        except ValueError as json_decode_error:
-            raise ParseResponseError(response_body, json_decode_error)
-        if self.server:
             if method == 'im.open':
                 if "ok" in result and result["ok"]:
                     self.server.attach_channel(kwargs["user"], result["channel"]["id"])
+            # If the call was for a `group` or `im` endpoint, attach the group data
```

Comment:
I've found that inline comments like this are some combination of three things:

1. A good sign to break out into a separate function
2. Suggest better naming of variables
3. Superfluous and potentially going to be out-of-date after a refactor

Reply:
Realized these could just reuse `server.parse_channel_data` is they were passed as a single-item array